internal/challenge/implme/intermediate/poke/app/app.go

```diff
@@ -73,8 +73,13 @@ func (p *pokeAPP) OnChanged(id string) {
 //
 // The function is currently not implemented and will panic if used.
 // TODO: Implement OnChangedNonBlocking to fetch and update Pokémon details asynchronously.
-func (p *pokeAPP) OnChangedNonBlocking(_ string) {
-	panic("implement me!")
+func (p *pokeAPP) OnChangedNonBlocking(id string) {
+	go func() {
```

Comment:
Same here, if you are creating go function I would expect you to make sure they are context aware and handling error correctly.